docker-compose.yml

```diff
@@ -71,7 +71,7 @@ services:
       BLACKFIRE_CLIENT_TOKEN:
     volumes:
       - '${REPO_PATH_SERVER}:/var/www/html'
-      - '${ADDITIONAL_APPS_PATH:-./data/apps-extra}:/var/www/html/apps-extra'
+      - '${ADDITIONAL_APPS_PATH:-./my-apps}:/var/www/html/apps-extra'
```

Comment:
Yes I agree with you that an application has several stable branches. But, we cannot run multiple server versions at the sale time as the project stands.
One more, we can switch between the branches.

As a developer, we can know very well if our application is made for this or that server version. After all depends on whether the developer is aware of this or not.

I think it would be better not to multiply the apps in the different server versions and to have a single point.

Finally, it all depends on how we orientate the operation of the project :slightly_smiling_face:

Reply:
> Yes I agree with you that an application has several stable branches. But, we cannot run multiple server versions at the sale time as the project stands.

We can, I use that daily and always have multiple versions running next to each other.

> As a developer, we can know very well if our application is made for this or that server version. After all depends on whether the developer is aware of this or not.

This still adds a lot of switching that I'd like to avoid.

What would work fine from my perspective is if you would still keep the separate apps-extra directories as they are but just add another app directory for apps that share a code base. Though I'm not really sure what the benefit of that would be.

Reply:
I am with @juliushaertl here. At least with the cookbook I have to maintain different versions (0.9.x for NC up to 24 and 0.10.x for NC 25). Having to migrate each time I switch the server version is unhandy at best. At worst there is the risk of killing the database structure with an incompatible version of the app and having to reset the dev environment of said app.

Reply:
I implemented this as part of https://github.com/juliushaertl/nextcloud-docker-dev/issues/104